e2e/cephfs.go

```diff
@@ -362,6 +369,274 @@ var _ = Describe("cephfs", func() {
+								name := fmt.Sprintf("%s%d", f.UniqueName, n)
+								err = createPVCAndApp(name, f, &p, &a, deployTimeout)
+								if err != nil {
+									e2elog.Logf("failed to create pvc and app %v", err)
+									Fail(err.Error())
+								}
+								w.Done()
+							}(&wg, i, *pvcClone, *appClone)
+						}
+						wg.Wait()
+
+						wg.Add(totalCount)
+						// delete clone and app
+						for i := 0; i < totalCount; i++ {
+							go func(w *sync.WaitGroup, n int, p v1.PersistentVolumeClaim, a v1.Pod) {
+								name := fmt.Sprintf("%s%d", f.UniqueName, n)
+								p.Spec.DataSource.Name = name
+								err = deletePVCAndApp(name, f, &p, &a)
+								if err != nil {
+									e2elog.Logf("failed to delete pvc and app %v", err)
+									Fail(err.Error())
+								}
+								w.Done()
+							}(&wg, i, *pvcClone, *appClone)
+						}
+						wg.Wait()
+
+						// create clones from different snapshosts and bind it to an
+						// app
+						wg.Add(totalCount)
+						for i := 0; i < totalCount; i++ {
+							go func(w *sync.WaitGroup, n int, p v1.PersistentVolumeClaim, a v1.Pod) {
+								name := fmt.Sprintf("%s%d", f.UniqueName, n)
+								p.Spec.DataSource.Name = name
+								err = createPVCAndApp(name, f, &p, &a, deployTimeout)
+								if err != nil {
+									e2elog.Logf("failed to create pvc and app %v", err)
+									Fail(err.Error())
+								}
+								w.Done()
+							}(&wg, i, *pvcClone, *appClone)
+						}
+						wg.Wait()
+
+						wg.Add(totalCount)
+						// delete snapshot
+						for i := 0; i < totalCount; i++ {
+							go func(w *sync.WaitGroup, n int, s v1beta1.VolumeSnapshot) {
+								s.Name = fmt.Sprintf("%s%d", f.UniqueName, n)
+								err = deleteSnapshot(&s, deployTimeout)
+								if err != nil {
+									e2elog.Logf("failed to delete snapshot %v", err)
+									Fail(err.Error())
+								}
+								w.Done()
+							}(&wg, i, snap)
+						}
+						wg.Wait()
+
+						wg.Add(totalCount)
+						// delete clone and app
+						for i := 0; i < totalCount; i++ {
+							go func(w *sync.WaitGroup, n int, p v1.PersistentVolumeClaim, a v1.Pod) {
+								name := fmt.Sprintf("%s%d", f.UniqueName, n)
+								p.Spec.DataSource.Name = name
+								err = deletePVCAndApp(name, f, &p, &a)
+								if err != nil {
+									e2elog.Logf("failed to delete pvc and app %v", err)
+									Fail(err.Error())
+								}
+								w.Done()
+							}(&wg, i, *pvcClone, *appClone)
+						}
+						wg.Wait()
+
+						// delete parent pvc
+						err = deletePVCAndValidatePV(f.ClientSet, pvc, deployTimeout)
+						if err != nil {
+							Fail(err.Error())
+						}
+					}
+				})
+
+				By("create a PVC-PVC clone and bind it to an app", func() {
+					v, err := f.ClientSet.Discovery().ServerVersion()
+					if err != nil {
+						e2elog.Logf("failed to get server version with error %v", err)
+						Fail(err.Error())
+					}
+					// pvc clone is only supported from v1.16+
+					if v.Major > "1" || (v.Major == "1" && v.Minor >= "16") {
```

Comment:
same here, check the inverse and `Skip()` the test, reduce indention to improve readability